sourcegraph/sourcegraph
# Svelte: Add new generic global navigation UI

Closes https://github.com/sourcegraph/sourcegraph/issues/61627

This PR adds a new navigation UI layout that is able to render any arbitrary UI from the children's level. This allows us to render different global navigation menu UI based on context 

- Global product nav on the home page
- Search bar on the search result page
- Repository navigation on the repo and blob UI pages


| Page | Screenshot |
| -------- | -------- |
| Search results with search bar | <img width="1230" alt="Screenshot 2024-04-18 at 16 11 23" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/14f453fd-ebdf-43a3-92b5-796b727d5ea8"> |
| Repository page with repo nav | <img width="1230" alt="Screenshot 2024-04-18 at 16 12 03" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/4300e2dc-816c-4d09-a63b-06bdfa4cbc5c"> |
| With opened sidebar nav | <img width="1230" alt="Screenshot 2024-04-18 at 16 12 33" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/5655a7cf-6537-4e24-a157-4bdd51b97b09"> |


## Test plan
- Go to the search home page, it should be plain navigation that we had prior to this PR
- Go to the search result home, check that navigation rendered search box instead of product navigation (product navigation is still available through sidebar navigation)
- Go to the repository page, and check that the main navigation renders the repository menu instead of the product menu
